Sources/MapboxMaps/Foundation/Tracing.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import os.signpost
+import MapboxCommon
+
+private let subsystem = "com.mapbox.maps"
+
+internal extension OSLog {
+    @available(iOS 12, *)
+    private static let _poi = OSLog(subsystem: subsystem, category: .pointsOfInterest)
+    private static let _platform = OSLog(subsystem: subsystem, category: "platform")
+    private static let tracingEnabled: Bool = SettingsServiceFactory.getInstance(storageType: .nonPersistent).isTracingEnabled
+
+    static let platform: OSLog = tracingEnabled ? _platform : .disabled
```

Comment:
BTW, is there a listener for settings service? We could adopt that one for reloading.

Reply:
I think it's not worth supporting the setting dynamically, especially considering that core maps reads this setting once. Mostly tracing is not an ad-hoc activity, so makes sense to have it static.
IMO using `SettingsServiceFactory` factory for enabling tracing is a bit deceiving, it gives the impression that tracing can be turned on/off at runtime, but that won't be respected.
Ideally we would indeed have it as an environment variable, e.g. `MAPBOX_MAPS_ENABLE_TRACING` that user can set in Xcode scheme.

Reply:
I think environment variable is insufficient for tracing functionality because it would limit debugging functionality with runs without Xcode. UserDefaults is a great mechanism that can be supported from the Scheme as well as leave an option to enable functionality persistently.
I'm agree with @evil159 that we can drop an idea of dynamic setting support.

Reply:
As we discussed with @persidskiy , there is no option to collect signpost metrics on non-debug run. 
Than env solution is totally fine for me.